src/modules/menu/components/advancedTools/ImportFile/utils.js

```diff
@@ -48,45 +48,60 @@ export async function handleFileContent(store, content, source, originalFile = n
     let layer = null
     let parsedContent
     let linkFiles
-    if (isZipContent(content)) {
-        log.debug(`File content is a zipfile, assume it is a KMZ archive`)
-        const kmz = await unzipKmz(content, source)
-        parsedContent = kmz.kml
-        linkFiles = kmz.files
-    } else if (isTiffContent(content)) {
-        log.debug(`File content might be a GeoTIFF, attempting a parse as such`)
+    if (content instanceof GeoTIFF) {
```

Comment:
Maybe this function could do with a little bit of refactoring, getting the handlings of the various types out in separate functions. It's pretty hard to read when it does what in what order and when it returns ^^.
Or at least the geotiff part, as from below, the code seems to assign and deal somehow `parsedContent`, whereas for this special case here this is irrelevant.

Reply:
Actually, come to think of it as I read `src/modules/menu/components/advancedTools/ImportFile/ImportFileOnlineTab.vue`, where it calls `handleFileContent` in two various ways: maybe we could have another function for handlingGeoTiff alltogether?